mongodbatlas.go

```diff
@@ -180,13 +196,23 @@ func (m *MongoDBAtlas) DeleteUser(ctx context.Context, req dbplugin.DeleteUserRe
 		}
 	}
 
-	// Default to "admin" if no db provided
-	if databaseUser.DatabaseName == "" {
-		databaseUser.DatabaseName = "admin"
+	// If the user is an X.509 user, delete the user from the X.509 certificate subject field
+	if strings.HasPrefix(req.Username, "CN=") {
```

Comment:
The comment is definitely helpful, so much that I think it might be worthwhile making a function out of it:
```go
func isX509User(username string) bool { ... }
```